### DISCUSSION TITLE
Inside Distrobox or not?

### DISCUSSION THREAD

**Discussion by doganulus** [89luca89/distrobox#1422] [General]
I like to have an alias declaration that works differently depending on my environment: 

```bash
# Host
alias code=flatpak run com.visualstudio.code
```

```bash
# Distrobox
alias code=distrobox-host-exec flatpak run com.visualstudio.code
```
Is there a stable environment variable you can suggest to distinguish whether I am in a distrobox? 

Normally I am looking for something like `DBX_VERSION` or equivalent. But open to other suggestions.

**Comment by doganulus** [comment]
Ok, I think an existence check for `/usr/bin/distrobox-host-exec` would work.

My script at `~/bin/code` is like that:
```bash
#!/bin/sh
if [ -x "/usr/bin/code" ]; then
    /usr/bin/code "$@"
elif [ -x "/usr/bin/flatpak" ] ; then
    /usr/bin/flatpak run com.visualstudio.code "$@"
elif [ -x "/usr/bin/distrobox-host-exec" ] ; then
    /usr/bin/distrobox-host-exec flatpak run com.visualstudio.code "$@"
else
    echo "Cannot run code"
    exit 255
fi
```

**Reply by sigzegv** [reply]
You should have an env var called DISTROBOX_ENTER_PATH when you are in the distrobox. could be cleaner than checking for a bin.
